### DISCUSSION TITLE
Modelling of high-contrast composite material

### DISCUSSION THREAD

**Discussion by MiguelVCarvalho** [Xiangyu-Hu/SPHinXsys#583] [Q&A] [answered]
Hello,

I would like to know if it is possible to solve the following problem with this code out-of-the-box or if it would be necessary to implement additional features (without considering the required input files).

The problem is as follows:

Consider a 2D planar rectangle composed of a matrix (white) and fiber material (green). The left side is fixed, and the right side has a prescribed displacement in the direction of the fibers equal to half of the rectangle's length (and is fixed in the perpendicular direction).

![image](https://github.com/Xiangyu-Hu/SPHinXsys/assets/25244490/389f879c-1511-4c84-a392-12b65572f47a)

This is proving to be a challenge when considering the finite element method because the composite involves materials with extremely different compliances: the fiber is approximately 10 orders of magnitude stiffer than the matrix (which is at 330°C). Considering this type of discretization, the extreme strain localization occurring in the matrix between the fibers leads to severely distorted elements (as shown below). Therefore, I am considering solving this problem using a meshless method, although I lack experience with this type of formulations.

**Initial configuration:**

![image](https://github.com/Xiangyu-Hu/SPHinXsys/assets/25244490/f7ba1745-5526-4ecc-a2eb-97eb05e5c18b)

**Final configuration (50% of elongation):**

![image](https://github.com/Xiangyu-Hu/SPHinXsys/assets/25244490/08d7936e-95d2-47b3-968f-485ae0735896)

In summary, I need to model a composite specimen composed of two elastic materials with extremely different stiffnesses under large strains to evaluate the feasibility of using a meshless method to solve this problem. If it works, the next step would be to model the matrix material as a viscoelastic material (with a given Prony series) but that can be analysed later on.

Thank you for your assistance.

**Comment by Xiangyu-Hu** [comment, marked as answer]
Do you mean 10 order of magnitude or magnitude of order 10? What is the the present issue when you are using fem for the simulation?

**Reply by Xiangyu-Hu** [reply]
Could you simply define the stiff material as rigid body?

**Reply by Xiangyu-Hu** [reply]
Could you simply define the stiff material as rigid body?

**Reply by MiguelVCarvalho** [reply]
Thank you for your response, @Xiangyu-Hu.

I meant 10 orders of magnitude. The fiber has a Young's modulus of around 270 GPa, while the matrix has an initial Young's modulus of 0.76 MPa, resulting in a contrast of about 5 orders of magnitude. However, when the matrix material is modeled as a viscous material (due to its temperature), the real stiffness is much lower than 0.76 MPa. Consequently, this stiffness contrast increases even further. The 10 orders of magnitude are an assumption because it is difficult to determine precisely in this case, but the matrix stiffness tends to zero because the long-term shear modulus is zero.

The main issue with modeling this in a FEM simulation is the element distortion for strains higher than 5% due to strain localization (as shown in the pictures above). This is the primary reason I want to try solving this with a mesh-free algorithm.

I haven't tried using rigid bodies for the fibers, but I would assume the mesh in the matrix material would still become completely distorted.

**Comment by Xiangyu-Hu** [comment]
@MiguelVCarvalho Although particle method has better isotropic property, it would require as test case to have a check as the maximum strain in this problem is very large. Since SPHinXsys is explicit method, modeling the fiber as rigid body ( actually very reasonable in this case) would leads to much faster simulation. As SPHinXsys is still in its development, the rigid body modeling now requires Simbody. However, this case may use simpler model which can be a simplified version from the present Simbody-coupling function.
